gen3utils/validate.py

```diff
@@ -0,0 +1,280 @@
+                ):
+                    should_check_has = (
+                        manifest_version(block_manifest["versions"], service_block)
+                        >= blocks_requirements[service_block]["version"]["min"]
+                        and manifest_version(block_manifest["versions"], service_block)
+                        < blocks_requirements[service_block]["version"]["max"]
+                    )
+                elif "min" in blocks_requirements[service_block]["version"]:
+                    should_check_has = (
+                        manifest_version(block_manifest["versions"], service_block)
+                        >= blocks_requirements[service_block]["version"]["min"]
+                    )
+                elif "max" in blocks_requirements[service_block]["version"]:
+                    should_check_has = (
+                        manifest_version(block_manifest["versions"], service_block)
+                        < blocks_requirements[service_block]["version"]["max"]
+                    )
+
+            if should_check_has:
+                # Validation to check if a service has a specific key in its block in cdis-manfiest
+                error_msg = (
+                    blocks_requirements[service_block]["has"]
+                    + " is missing in "
+                    + service_block
+                    + " block or "
+                    + service_block
+                    + " block is missing"
+                )
+                ok = (
+                    assert_and_log(
+                        service_block in block_manifest
+                        and blocks_requirements[service_block]["has"]
+                        in block_manifest[service_block],
+                        error_msg,
+                    )
+                    and ok
+                )
+
+            if blocks_requirements[service_block] == "true":
+                # Validation to check if a block exists in cdis-manifest
+                ok = (
+                    assert_and_log(
+                        service_block in block_manifest,
+                        service_block + " block is missing in cdis-manifest",
+                    )
+                    and ok
+                )
+    return ok
+
+
+def manifest_version(manifest_versions, services):
+    """
+        Arg: 
+            services:microservice name
+            manifest_versions: the versions block from manifest.json
+        Return:
+            microservice version
+    """
+    for manifest_version in manifest_versions:
+        if manifest_version == services:
+            # if this fails, make sure you check the service is in the manifest before
+            # calling manifest_version()! it should never happen
+            microservice_version = manifest_versions.get(services).split(":")[1]
+            if "_" in microservice_version:
+                logger.warning(services + " is on branch")
```

Comment:
what happens if the service is on a branch? the version is still returned so i think the > and < comparisons would behave weirdly